Volume/Views/BrowserView.swift

```diff
@@ -15,139 +15,197 @@ import WebKit
+            .sink { completion in
+                if case let .failure(error) = completion {
+                    print(error.localizedDescription)
                 }
-
-                Spacer()
-                    .frame(width: 16)
-
-                Button {
-                    incrementShoutouts()
-                } label: {
-                    Image("shout-out")
-                        .resizable()
-                        .scaledToFit()
-                        .frame(height: 24)
-                        .foregroundColor(isShoutoutsButtonEnabled ? Color.volume.orange : Color.gray)
+            } receiveValue: { article in
+                if let fields = article.article?.fragments.articleFields {
+                    state = .results(Article(from: fields))
                 }
-                .disabled(!isShoutoutsButtonEnabled)
-
-                Spacer()
-                    .frame(width: 6)
-
-                Text(String(max(article.shoutouts, userData.shoutoutsCache[article.id, default: 0])))
-                    .font(.latoRegular(size: 12))
             }
-        }
-        .padding([.leading, .trailing], 16)
-        .padding([.top, .bottom], 8)
-        .background(Color.volume.backgroundGray)
     }
-
-    func displayShareScreen() {
+    
+    func displayShareScreen(for article: Article) {
         let rawString = Secrets.openArticleUrl + article.id
         if let shareArticleUrl = URL(string: rawString) {
             let linkSource = LinkItemSource(url: shareArticleUrl, article: article)
             let shareVC = UIActivityViewController(activityItems: [linkSource], applicationActivities: nil)
             UIApplication.shared.windows.first?.rootViewController?.present(shareVC, animated: true)
         }
     }
+}
 
-    var body: some View {
-        VStack(spacing: 0) {
-            if let articleUrl = article.articleUrl {
-                WebView(url: articleUrl)
-                    .onAppear {
-                        AppDevAnalytics.log(VolumeEvent.openArticle.toEvent(.article, value: article.id, navigationSource: navigationSource))
-                    }
-                    .onDisappear {
-                        AppDevAnalytics.log(VolumeEvent.closeArticle.toEvent(.article, value: article.id, navigationSource: navigationSource))
-                    }
-            }
-            toolbar
-        }.toolbar {
-            ToolbarItem(placement: .principal) {
-                VStack(spacing: 0) {
-                    Text(article.publication.name)
-                        .font(.begumBold(size: 12))
-                    Text("Reading in Volume")
-                        .font(.latoRegular(size: 10))
-                        .foregroundColor(Color.volume.lightGray)
-                }
-            }
-        }
+extension BrowserView {
+    private enum BrowserViewState<Results> {
+        case loading, results(Results)
+    }
+    
+    public enum BrowserViewInitType {
```

Comment:
Why public?

Reply:
This enum is required to initialize `BrowserView`

Reply:
Yeah but why are you explicitly using the public keyword? It should be internal by default, and we don't need it accessible from other frameworks.